lib/ruby_lsp/listeners/spec_style.rb

```diff
@@ -0,0 +1,200 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Listeners
+    class SpecStyle
+      extend T::Sig
+      include Requests::Support::Common
+
+      DYNAMIC_REFERENCE_MARKER = "<dynamic_reference>"
+
+      sig do
+        params(
+          response_builder: ResponseBuilders::TestCollection,
+          global_state: GlobalState,
+          dispatcher: Prism::Dispatcher,
+          uri: URI::Generic,
+        ).void
+      end
+      def initialize(response_builder, global_state, dispatcher, uri)
+        @response_builder = response_builder
+        @current_group = T.let(nil, T.nilable(Requests::Support::TestItem))
+        @uri = uri
+        @index = T.let(global_state.index, RubyIndexer::Index)
+
+        @visibility_stack = T.let([:public], T::Array[Symbol])
+        @nesting = T.let([], T::Array[String])
+        @test_group_nesting = T.let([], T::Array[String])
+        @spec_stack = T.let([], T::Array[T::Boolean])
+
+        dispatcher.register(
+          self,
+          :on_class_node_enter,
+          :on_class_node_leave,
+          :on_module_node_enter,
+          :on_module_node_leave,
+          :on_call_node_enter,
+          :on_call_node_leave,
+          :on_def_node_enter,
+        )
+      end
+
+      sig { params(node: Prism::ClassNode).void }
+      def on_class_node_enter(node)
+        @visibility_stack << :public
+        name = constant_name(node.constant_path)
+        name ||= name_with_dynamic_reference(node.constant_path)
+
+        fully_qualified_name = RubyIndexer::Index.actual_nesting(@nesting, name).join("::")
+
+        attached_ancestors = begin
+          @index.linearized_ancestors_of(fully_qualified_name)
+        rescue RubyIndexer::Index::NonExistingNamespaceError
+          # When there are dynamic parts in the constant path, we will not have indexed the namespace. We can still
```

Comment:
Do we have a text covering this?

Reply:
We do have tests for this. It's the ones with stuff like `var::FooTest`.